lstchain/reco/reconstructorCC.py

```diff
@@ -293,12 +358,26 @@ def log_pdf(charge, t_cm, x_cm, y_cm, length, wl, psi, v, rl,
     dy = (p_y - y_cm)
     long = dx * np.cos(psi) + dy * np.sin(psi)
     t_model = linval(v, t_cm, long)
-    t = np.empty(data.shape, dtype=np.float64)
-    for i in range(n_pixels):
-        for j in range(n_samples):
-            t[i, j] = times[j] - t_model[i] - time_shift[i]
-    templates = template_interpolation(is_high_gain, t, template_t0, template_dt,
-                                       template_hg, template_lg)
+
+    t_model_datashape = t_model[:, None] * (np.ones((n_pixels, n_samples), dtype=t_model.dtype))
+    time_shift_datashape = time_shift[:, None] * (np.ones((n_pixels, n_samples), dtype=time_shift.dtype))
+    t_0 = t_model_datashape + time_shift_datashape
+
+    # Use template_interpolation_withNSB to interpolate without adding NSB. Consider additional_nsb
+    # as the number of samples in each pixel, no previous values in the waveform and constant amplitude of 1.
+    templates = template_interpolation_withNSB(
+        empty_waveform=np.zeros(data.shape),
+        time=times,
+        is_high_gain=is_high_gain,
+        additional_nsb=np.full(n_pixels, n_samples),
+        amplitude=np.ones(data.shape),
+        t_0=t_0,
+        t0_template=template_t0,
+        dt_template=template_dt,
+        a_hg_template=template_hg,
+        a_lg_template=template_lg
+    )
+
```

Comment:
I am not convinced by changing the call in log_pdf.
The loop before the function is already in njit code. You add multiple operations before the call to the interpolation. And the code inside is in my opinion not faster. (Would need to test).

Reply:
You are right, even though if I use ` t_0 = (time_shift + t_model)[:, None] * (np.ones((n_pixels, n_samples), dtype=np.float64))` the loop format in njit code is faster than the numpy product. t=4.8e-05 s and 3.3e-05 s, respectively. 
Using `template_interpolation` in `log_pdf` is fast, clear and tested. I will undo the changes in `log_pdf`